src/meta/meta_turtle.c

```diff
@@ -81,12 +94,76 @@ __metadata_load_hot_backup(WT_SESSION_IMPL *session)
         WT_ERR(__wt_getline(session, fs, value));
         if (value->size == 0)
             WT_ERR_PANIC(session, EINVAL, "%s: zero-length value", WT_METADATA_BACKUP);
+        /*
+         * When performing partial backup restore, check through the file metadata entries and check
+         * if the file exists in the file system. If the file doesn't exist, append the filename to
+         * the partial backup list and the partial backup remove list so we can clean up the
+         * metadata and history store afterwards.
+         */
+        filename = (char *)key->data;
+        if (restore_partial_backup && WT_PREFIX_SKIP(filename, "file:")) {
+            WT_ERR(__wt_fs_exist(session, filename, &exist));
+            if (!exist) {
+                WT_ERR(__wt_realloc_def(
+                  session, &allocated_int, slot + 1, &conn->partial_backup_remove_list));
+                /* Leave a NULL at the end to mark the end of the list. */
+                WT_ERR(__wt_realloc_def(session, &allocated, slot + 2, &partial_backup_list));
+                p = &partial_backup_list[slot];
+                p[0] = p[1] = NULL;
+
+                WT_ERR(__wt_strdup(session, (char *)key->data, p));
+                WT_ERR(__wt_config_getones(session, (char *)value->data, "id", &cval));
+                conn->partial_backup_remove_list[slot] = (uint32_t)cval.val;
+                slot++;
+            }
+        } else if (restore_partial_backup && WT_SUFFIX_MATCH(filename, ".wti") &&
+          WT_SUFFIX_MATCH(filename, ".lsm")) {
+            WT_ERR_MSG(session, EINVAL,
+              "%s: partial backup currently doesn't support index or lsm files.", filename);
+        }
         WT_ERR(__wt_metadata_update(session, key->data, value->data));
```

Comment:
The fact that we're adding the non-existent table to the metadata deserves a comment here about "Add the entry to the metadata even if it doesn't exist so that we can correctly drop all related entries via the schema code later."